interface/billing/billing_report.php

```diff
@@ -139,32 +139,32 @@
         function doSubmit(action) {
             top.restoreSession();
             return new Promise(function(resolve, reject) {
-                if (action !== 'btn-continue') {
-                    var showLog = function() {
-                        $("#view-log-link").click();
-                    };
-                    // Pre-open a dialog and target dialogs iframe for content from billing_process
-                    // text or PDF.
-                    dlgopen('', 'ValidateShowBatch', 875, 500, false, '', {
-                        buttons: [{
-                                text: '<?php echo xlt("Logs"); ?>',
-                                close: false,
-                                style: 'default btn-sm',
-                                click: showLog
-                            },
-                            {
-                                text: '<i class="fa fa-thumbs-up"></i>&nbsp;<?php echo xlt("Close"); ?>',
-                                close: true,
-                                style: 'default btn-sm'
-                            }
-                        ],
-                        //onClosed: 'SubmitTheScreen', // future and/or example of onClosed.
-                        sizeHeight: 'full'
-                    });
-                    // target content from submit to dialogs iframe
-                    document.update_form.target = 'ValidateShowBatch';
-                }
+                var showLog = function() {
+                    $("#view-log-link").click();
+                };
+                // Pre-open a dialog and target dialogs iframe for content from billing_process
+                // text or PDF.
+                dlgopen('', 'ValidateShowBatch', 875, 500, false, '', {
+                    buttons: [{
+                            text: '<?php echo xlt("Logs"); ?>',
+                            close: false,
+                            style: 'default btn-sm',
+                            click: showLog
+                        },
+                        {
+                            text: '<i class="fa fa-thumbs-up"></i>&nbsp;<?php echo xlt("Close"); ?>',
+                            close: true,
+                            style: 'default btn-sm'
+                        }
+                    ],
+                    //onClosed: 'SubmitTheScreen', // future and/or example of onClosed.
+                    sizeHeight: 'full'
+                });
+                // target content from submit to dialogs iframe
+                document.update_form.target = 'ValidateShowBatch';
+
                 // Now submit form and populate dialog.
+                top.restoreSession(); // Not sure if this is needed but something in billing is causing 'SITE ID' error
```

Comment:
never can go wrong with this :)
(as in this case, it should always be called before a submit/redirect/call go to another script to ensure that the correct cookie id is set)
(this is probably not the reason for the SITE ID error though)